shared/modules/macros.jinja

```diff
@@ -88,3 +88,80 @@ ocil: |
 
 {{{ audit_syscall_check_ocil_clause_entry() }}}
 {{%- endmacro %}}
+
+
+{{%- macro _firewalld_check(access_action, port, proto, service) %}}
+    To determine if <code>firewalld</code> is configured to {{{ access_action }}} to <code>{{{ service }}}</code>
+    on port <code>{{{ port }}}/{{{ proto }}}</code>, run the following command(s):
+    {{% if port %}}
+        <code>firewall-cmd --list-ports</code>
+    {{% endif %}}
+    {{% if service %}}
+        <code>firewall-cmd --list-services</code>
+    {{% endif %}}
+{{%- endmacro %}}
+
+
+{{%- macro firewalld_allow_access(port, proto, service) %}}
+{{{ _firewalld_check("allow access", port, proto, service) }}}
+    If <code>firewalld</code> is configured to allow access through the firewall, something similar to the following will be output:
+    {{% if service %}}
+    If it is a service:
+    <code>&gt;{{{ service }}}</code>
+    {{% endif %}}
+    {{% if port %}}
+    If it is a port:
+    <code>{{{ port }}}/{{{ proto }}}</code>
+    {{% endif %}}
+{{%- endmacro %}}
+
+
+{{%- macro firewalld_prevent_access(port, proto, service) %}}
+{{{ _firewalld_check("prevent access", port, proto, service) }}}
+    If <code>firewalld</code> is configured to prevent access, no output will be returned.
+{{%- endmacro %}}
+
+
+{{%- macro firewalld_prevent_desc(port, proto, service) %}}
```

Comment:
should probably be two macros, one for port/proto and one for service

Reply:
That would make four macros instead of two, plus the current code allows and meaningfully handles the case when all arguments are supplied. Do we really want another split?

Reply:
maybe not, I didn't read it properly